static/js/timerender.ts

```diff
@@ -24,6 +24,80 @@ export function clear_for_testing(): void {
     next_timerender_id = 0;
 }
 
+type DateFormat = "short" | "medium" | "day" | "full";
+type DateWithTimeFormat = "shortWithTime" | "mediumWithTime" | "longWithTime";
+type TimeFormat = "time";
+
+type DateOrTimeFormat = DateFormat | TimeFormat | DateWithTimeFormat;
+
+// The function takes in an optional type parameter and returns
+// an options object that can be used to format a date/time using the Intl.DateTimeFormat API.
+// This function allows easy and consistent formatting of dates and times
+// with different options to customize the format according to the specified type
+function get_format_options_for_type(type: DateOrTimeFormat): Intl.DateTimeFormatOptions {
+    const isHourCycleTwentyFourHour = user_settings.twenty_four_hour_time;
+
+    const timeFormatOptions: Intl.DateTimeFormatOptions = isHourCycleTwentyFourHour
+        ? {hourCycle: "h23", hour: "2-digit", minute: "2-digit"}
+        : {
+              hourCycle: "h12",
+              hour: "numeric",
+              minute: "2-digit",
+          };
+
+    const dayFormatOptions: Intl.DateTimeFormatOptions = {weekday: "long"};
+    const fullFormatOptions: Intl.DateTimeFormatOptions = {dateStyle: "full"};
```

Comment:
Our naming convention for variables in our own codebase is to use underscores, not `camelCase`. The motivation for that is our server and API all use underscores in property names, and having a consistent style makes grepping for things like `twenty_four_hour_time` a better experience than having the frontend rename it to `twentyFourHourTime` as a coding convention.

I'm not sure what is best in this function, which is pretty coupled to the the `Intl` library, but I think it'd probably be better to be consistent with our coding pattern, at least for the parts that are exposed externally, like `shortWithTime`, and possibly just everywhere.

Reply:
Got it!

When working on the codebase, I tried to be consistent with the existing naming conventions, but I noticed that there are both camel and snake case used. 
I haven't [found](https://zulip.readthedocs.io/en/latest/search.html?q=naming&check_keywords=yes&area=default#) any documentation about naming conventions for the frontend part of the app. Should we document these agreements better in the [docs](https://zulip.readthedocs.io/en/latest/contributing/code-style.html#code-style-and-conventions)? I believe that being explicit is better than being implicit.

Additionally, should we consider adding [naming-convention](https://typescript-eslint.io/rules/naming-convention/) rule in eslint? I understand there may be reasons not to do so, but it could be beneficial as it would eliminate the need to check variable naming in PRs and allow us to focus on more important things.

Reply:
> I noticed that there are both camel and snake case used.

Actually, it was an `date-fns` library method. ;)

Reply:
Yeah, a naming-convention `eslint` rule would likely be a good idea. I generally a prefer a linter to more docs where possible, because it's hard to read all the documentation when joining a new project, and also tools will do things like autofix stuff for you.

Adding that may require a bit of care to figure out the right configuration, since there's third-party stuff we don't control, etc; you can start a #frontend thread about it.